src/PolygonWorldID.sol

```diff
@@ -51,23 +148,28 @@ contract PolygonWorldID is WorldIDBridge, FxBaseChildTunnel, Ownable {
         override
         validateSender(sender)
     {
-        // This decodes as specified in the parameter block. If this fails, it will revert.
-        (uint256 newRoot, uint128 timestamp) = abi.decode(message, (uint256, uint128));
+        bytes4 selector = grabSelector(message);
+        bytes memory payload = stripSelector(message);
 
-        _receiveRoot(newRoot, timestamp);
+        if (selector == receiveRootSelector) {
+            (uint256 root, uint128 timestamp) = abi.decode(payload, (uint256, uint128));
+            _receiveRoot(root, timestamp);
+        } else if (selector == receiveRootHistoryExpirySelector) {
+            uint256 rootHistoryExpiry = abi.decode(payload, (uint256));
+            _setRootHistoryExpiry(rootHistoryExpiry);
+        } else {
+            revert InvalidMessageSelector();
+        }
     }
 
     ///////////////////////////////////////////////////////////////////////////////
     ///                              DATA MANAGEMENT                            ///
     ///////////////////////////////////////////////////////////////////////////////
 
-    /// @notice Sets the amount of time it takes for a root in the root history to expire.
-    ///
-    /// @param expiryTime The new amount of time it takes for a root to expire.
-    ///
-    /// @custom:reverts string If the caller is not the owner.
-    function setRootHistoryExpiry(uint256 expiryTime) public virtual override onlyOwner {
-        _setRootHistoryExpiry(expiryTime);
+    /// @notice Placeholder to satisfy WorldIDBridge inheritance
+    /// @dev This function is not used on Polygon PoS because of FxPortal message passing architecture
+    function setRootHistoryExpiry(uint256) public virtual override {
+        revert SetRootHistoryExpiryPlaceholder();
```

Comment:
Here's one case where I would actually revert with a string. Something like `"Root history expiry should only be set via the state bridge"`.